src/App/Pages/Settings/SettingsPage/SettingsPageViewModel.cs

```diff
@@ -257,6 +257,10 @@ public async Task VaultTimeoutAsync(bool promptOptions = true, int? newTimeout =
                 }
                 var cleanSelection = selection.Replace("✓ ", string.Empty);
                 var selectionOption = _vaultTimeouts.FirstOrDefault(o => o.Key == cleanSelection);
+                if (selectionOption.Value == null && selectionOption.Value != oldTimeout)
+                {
+                    await _platformUtilsService.ShowDialogAsync(AppResources.NeverLockWarning, null, AppResources.Ok);
```

Comment:
Even though this is fine, I think we should keep the same pattern as when selecting "Log out" as vault timeout action, where we have the title "Warning" and the ability to cancel the change which makes it go back to the previous value

Reply:
Agreed @fedemkr, I will change the dialog to have that behaviour.